src/Symfony/Component/PropertyAccess/Tests/PropertyAccessorTest.php

```diff
@@ -479,6 +479,19 @@ public function getValidPropertyPaths()
         ];
     }
 
+    public function getValidReadPropertyPaths()
+    {
+        $testCases = $this->getValidWritePropertyPaths();
+
+        // Optional paths can only be read and can't be written to.
+        $testCases[] = [(object) ['foo' => (object) ['firstName' => 'Bernhard']], 'foo.bar?', null];
+        $testCases[] = [(object) ['foo' => (object) ['firstName' => 'Bernhard']], 'foo.bar?.baz', null];
```

Comment:
Actually, `foo` has no `bar` property here. I'd expect it to fail, not to return `null`. 
`null` would be expected if and only if `foo.bar` exists but is `null` instead of an object, hence cannot read the `bar.baz` value.

Reply:
🤔 ReReading the RFC and the original post, I indeed got it completely wrong, 

I will recheck it, 

thanks